### PR TITLE
Add ERC721 and ERC1155 connector

### PR DESCRIPTION
**Pull request by cryptoDev222**
Implement ERC721 deposit/withdraw function

### CODE REVIEW

File: contracts/mainnet/connectors/basic-ERC721/main.sol [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        tokenContract.safeTransferFrom(address(this), to, _tokenId);
+
+        setUint(setId, _tokenId);
+
+        _eventName = "LogWithdrawERC721(address,uint256,address,uint256,uint256)";
+        _eventParam = abi.encode(token, _tokenId, to, getId, setId);
+    }
+}
+
+contract ConnectV2Basic is BasicResolver {
```
Comment (thrilok209) [inline contracts/mainnet/connectors/basic-ERC721/main.sol]:
```suggestion
contract ConnectV2BasicERC721 is BasicResolver {
```

File: contracts/mainnet/connectors/basic-ERC721/main.sol [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    string public constant name = "BASIC-ERC721-A";
```
Comment (thrilok209) [inline contracts/mainnet/connectors/basic-ERC721/main.sol]:
```suggestion
    string public constant name = "BASIC-ERC721-v1.0";
```

File: contracts/mainnet/connectors/basic-ERC721/main.sol [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import {DSMath} from "../../common/math.sol";
+import {Basic} from "../../common/basic.sol";
+import {Events} from "./events.sol";
+
+abstract contract BasicResolver is Events, DSMath, Basic {
+    /**
+     * @dev Deposit Assets To Smart Account.
+     * @notice Deposit a ERC721 token to DSA
+     * @param token The address of the token to deposit.
+     * @param tokenId The id of token to deposit.
```
Comment (thrilok209) [inline contracts/mainnet/connectors/basic-ERC721/main.sol]: @cryptoDev222 Could you update the comments?

File: contracts/mainnet/connectors/basic-ERC721/main.sol (hunk 1 above)
Comment (cryptoDev222) [reply contracts/mainnet/connectors/basic-ERC721/main.sol:74]: Okay, got it.

File: contracts/mainnet/connectors/basic-ERC721/main.sol (hunk 3 above)
Comment (cryptoDev222) [reply contracts/mainnet/connectors/basic-ERC721/main.sol:18]: Okay, got it.

File: contracts/mainnet/connectors/basic-ERC1155/main.sol [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        uint256 getId,
+        uint256 setId
+    )
+        public
+        payable
+        returns (string memory _eventName, bytes memory _eventParam)
+    {
+        uint256 _amount = getUint(getId, amount);
+        IERC1155 tokenContract = IERC1155(token);
+        tokenContract.safeTransferFrom(address(this), to, tokenId, _amount, "");
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/basic-ERC1155/main.sol:82]: on withdraw it'll always be `safeTransfer`

File: contracts/mainnet/connectors/basic-ERC721/main.sol [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        uint256 getId,
+        uint256 setId
+    )
+        public
+        payable
+        returns (string memory _eventName, bytes memory _eventParam)
+    {
+        uint256 _tokenId = getUint(getId, tokenId);
+        IERC721 tokenContract = IERC721(token);
+        tokenContract.safeTransferFrom(address(this), to, _tokenId);
```
Comment (KaymasJain) [inline contracts/mainnet/connectors/basic-ERC721/main.sol:65]: Same here `safeTransfer`

File: contracts/mainnet/connectors/basic-ERC1155/main.sol (hunk 4 above)
Comment (cryptoDev222) [reply contracts/mainnet/connectors/basic-ERC1155/main.sol:82]: There is no safeTransfer function on IERC1155.

File: contracts/mainnet/connectors/basic-ERC721/main.sol (hunk 5 above)
Comment (cryptoDev222) [reply contracts/mainnet/connectors/basic-ERC721/main.sol:65]: There is no safeTransfer function on IERC1155.

**Comment by thrilok209** [comment]
### Deployed connector:

**Connector Name**: "BASIC-B"
**Connector Address**: [0x4DfA1780ae85f0Ec7197c61ffe533C7dc84F15e9](https://etherscan.io/address/0x4DfA1780ae85f0Ec7197c61ffe533C7dc84F15e9)

**Connector Name**: "BASIC-C"
**Connector Address**: [0x6ECbf5A77D65f857C66FE729cb3CD7835369c42B](https://etherscan.io/address/0x6ECbf5A77D65f857C66FE729cb3CD7835369c42B)

**Comment by thrilok209** [comment]
"BASIC-B" and "BASIC-C" connector whitelisted: https://etherscan.io/tx/0xfa0a651fae37ffa0ced672d79b16fc015fd50d04b7a35176642e68dc228fc0bb

**Review by thrilok209 — APPROVED** [review]

**Pull request merged**